ui/pages/confirmations/components/confirm/info/shared/redesigned-gas-fees/use-fees-calculations.ts

```diff
@@ -0,0 +1,167 @@
+
+      // L2
+      const feesL2 = getFeesFromHex(gasEstimate);
+      setCurrentCurrencyL2Fee(feesL2.currentCurrencyFee);
+      setNativeCurrencyL2Fee(feesL2.nativeCurrencyFee || defaultFee);
+
+      // Estimated fee from breakdown = L1 + L2
+      const estimatedFeeFromBreakdown = addHexes(gasEstimate, layer1GasFee);
+      const feesFromBreakdown = getFeesFromHex(estimatedFeeFromBreakdown);
+      setCurrentCurrencyEstimatedFeeFromBreakdown(
+        feesFromBreakdown.currentCurrencyFee,
+      );
+      setNativeCurrencyEstimatedFeeFromBreakdown(
+        feesFromBreakdown.nativeCurrencyFee || defaultFee,
+      );
+    }
+  }, [hasLayer1GasFee, layer1GasFee, gasEstimate, defaultFee]);
+
+  useEffect(() => {
+    if (!hasLayer1GasFee) {
+      // Estimated fee without breaking down L1 and L2 fees
+      const gasLimit =
+        transactionMeta?.dappSuggestedGasFees?.gas ||
+        transactionMeta?.txParams?.gas ||
+        '0x';
+      const gasPrice =
+        transactionMeta?.dappSuggestedGasFees?.gasPrice ||
+        transactionMeta?.txParams?.gasPrice ||
+        '0x';
+
+      let estimatedFee;
+      if (supportsEIP1559) {
+        // Minimum Total Fee (EIP1559) = (estimatedBaseFee + maxPriorityFeePerGas) * gasLimit
+        const minimumFeePerGas = addHexes(
+          transactionMeta?.txParams?.estimatedBaseFee || '0x0',
+          maxPriorityFeePerGas,
+        );
+
+        estimatedFee = multiplyHexes(minimumFeePerGas, gasLimit);
+      } else {
+        // Total Fee (Type 0) = gasPrice * gasLimit
+        estimatedFee = multiplyHexes(gasPrice, gasLimit);
+      }
+
+      const estimatedFees = getFeesFromHex(estimatedFee);
+
+      setCurrentCurrencyEstimatedFee(estimatedFees.currentCurrencyFee);
+      setNativeCurrencyEstimatedFee(
+        estimatedFees.nativeCurrencyFee || defaultFee,
+      );
+    }
+  }, [
+    hasLayer1GasFee,
+    supportsEIP1559,
+    transactionMeta,
+    maxPriorityFeePerGas,
+    defaultFee,
+  ]);
+
+  const gasLimit =
+    transactionMeta?.dappSuggestedGasFees?.gas ||
+    transactionMeta?.txParams?.gas ||
+    '0x';
+  const gasPrice = transactionMeta?.txParams?.gasPrice || '0x';
+
+  const maxFee = useMemo(() => {
+    if (supportsEIP1559) {
+      // Max fee = maxFeePerGas * gasLimit
+      return multiplyHexes(maxFeePerGas, gasLimit);
+    }
+
+    // Total Fee (Type 0) = gasPrice * gasLimit
+
+    return multiplyHexes(gasPrice, gasLimit);
+  }, [supportsEIP1559, maxFeePerGas, transactionMeta]);
+
+  const {
+    currentCurrencyFee: currentCurrencyMaxFee,
+    nativeCurrencyFee: nativeCurrencyMaxFee,
+  } = getFeesFromHex(maxFee);
+
+  return {
+    currentCurrencyEstimatedFeeFromBreakdown,
+    currentCurrencyEstimatedFee,
+    nativeCurrencyEstimatedFeeFromBreakdown,
+    nativeCurrencyEstimatedFee,
+    currentCurrencyL1Fee,
+    nativeCurrencyL1Fee,
+    currentCurrencyL2Fee,
+    nativeCurrencyL2Fee,
+    currentCurrencyMaxFee,
+    nativeCurrencyMaxFee,
```

Comment:
Rather than putting the most unique aspect at the end of the name, would it be more readable to call these `l1FeeCurrency`, `l1FeeNative`, `maxFeeCurrency`, `maxFeeNative` etc?